keps/2349-multikueue-external-custom-job-support/README.md

```diff
@@ -0,0 +1,247 @@
+
+## Summary
+This KEP introduces a mechanism to support MultiKueue for custom Jobs. This is achieved by introducing a configurable system that allows administrators to choose between a built-in generic MultiKueue adapter or an external controller for managing custom job types.
+
+## Motivation
+
+Many Kueue adopters have specific CustomResources which might not be public.
+Kueue should give a possibility to manage such custom Jobs across multiple clusters.
+
+### Goals
+
+Give a possibility for external controllers to manage custom Jobs in a MultiKueue environment.
+
+
+### Non-Goals
+
+## Proposal
+
+This KEP proposes extending MultiKueue to support custom jobs through a configurable mechanism. Administrators will be able to define, for each custom job type (identified by its GroupVersionKind), whether it should be handled by a built-in generic adapter or by an external, multi-cluster-aware controller.
+
+This configuration will be managed via a configmap. Based on this configuration, MultiKueue will either use its generic adapter for a given GVK or expect an external controller to manage the job's lifecycle across clusters, in which case MultiKueue will only be responsible for synchronizing the `Workload` object.
+
+### User Stories
+
+As a cluster administrator I would like to have the ability to schedule Jobs
+which aren't natively supported by Kueue, eg Tekton.
+
+### Constraints
+
+- Fixed Transformation: The generic adapter is limited to a fixed, non-configurable transformation logic.
+
+### Risks and Mitigations
+
+#### Risks
+
+ Configmaps are unstructured by default. They don't provide validation by default unlike CRD. User could make typo mistakes or give invalid syntax in configmap.
+
+#### Mitigations
+
+Configmaps that are not correct should be logged with errors like "Invalid GVK format". We could have a CLI command to validate the adapter logic.
+
+## Design Details
+
+### Configurable MultiKueue Adapters
+
+To support external frameworks, the MultiKueue configuration API will be extended. The following definitions will be added to the `apis/config/v1beta1` package:
+
+```go
+// MultiKueue specifies the multikueue configuration options.
+type MultiKueue struct {
+   .....
+	// ExternalFrameworks lists the external frameworks to be supported by MultiKueue.
+	// +optional
+	ExternalFrameworks []ExternalFramework `json:"externalFrameworks,omitempty"`
+}
+
+// ExternalFramework defines a framework that is not built-in.
+type ExternalFramework struct {
+	// Group is the API group of the custom job.
+	// +required
+	Group string `json:"group"`
+	// Version is the API version of the custom job.
+	// +required
+	Version string `json:"version"`
+	// Kind is the Kind of the custom job.
+	// +required
+	Kind string `json:"kind"`
+	// Controller specifies which controller should manage this custom job.
+	// Can be "generic" or "external".
+	// Defaults to "generic".
+	// +optional
+	Controller ControllerType `json:"controller,omitempty"`
```

Comment:
I would suggest dropping this for now, we can introduce when proven this is needed.